drivers/npmx/npmx_shell.c

```diff
@@ -3200,19 +3487,72 @@ SHELL_STATIC_SUBCMD_SET_CREATE(sub_vbusin_status,
+			       SHELL_CMD(get, NULL, "Get inverted polarity status",
+					 cmd_ship_config_inv_polarity_get),
+			       SHELL_CMD(set, NULL, "Set inverted polarity status",
+					 cmd_ship_config_inv_polarity_set),
 			       SHELL_SUBCMD_SET_END);
 
+/* Creating subcommands (level 3 command) array for command "ship config". */
+SHELL_STATIC_SUBCMD_SET_CREATE(sub_ship_config,
+			       SHELL_CMD(time, &sub_ship_config_time, "Time", NULL),
+			       SHELL_CMD(inv_polarity, &sub_ship_config_inv_polarity,
+					 "Button invert polarity", NULL),
+			       SHELL_SUBCMD_SET_END);
+
+/* Creating subcommands (level 4 command) array for command "ship reset long_press". */
+SHELL_STATIC_SUBCMD_SET_CREATE(sub_ship_reset_long_press,
+			       SHELL_CMD(get, NULL, "Get long press status",
+					 cmd_ship_reset_long_press_get),
+			       SHELL_CMD(set, NULL, "Set long press status",
+					 cmd_ship_reset_long_press_set),
+			       SHELL_SUBCMD_SET_END);
+
+/* Creating subcommands (level 4 command) array for command "ship reset two_buttons". */
+SHELL_STATIC_SUBCMD_SET_CREATE(sub_ship_reset_two_buttons,
+			       SHELL_CMD(get, NULL, "Get two buttons status",
+					 cmd_ship_reset_two_buttons_get),
+			       SHELL_CMD(set, NULL, "Set two buttons status",
+					 cmd_ship_reset_two_buttons_set),
+			       SHELL_SUBCMD_SET_END);
+
+/* Creating subcommands (level 3 command) array for command "ship reset". */
+SHELL_STATIC_SUBCMD_SET_CREATE(
+	sub_ship_reset, SHELL_CMD(long_press, &sub_ship_reset_long_press, "Long press", NULL),
+	SHELL_CMD(two_buttons, &sub_ship_reset_two_buttons, "Two buttons", NULL),
+	SHELL_SUBCMD_SET_END);
+
+/* Creating subcommands (level 3 command) array for command "ship mode". */
+SHELL_STATIC_SUBCMD_SET_CREATE(sub_ship_mode,
+			       SHELL_CMD(ship, NULL, "Enter ship mode", cmd_ship_mode_ship_set),
+			       SHELL_CMD(hibernate, NULL, "Enter hibernate mode",
+					 cmd_ship_mode_hibernate_set),
+			       SHELL_SUBCMD_SET_END);
+
+/* Creating subcommands (level 2 command) array for command "ship". */
+SHELL_STATIC_SUBCMD_SET_CREATE(sub_ship, SHELL_CMD(config, &sub_ship_config, "Ship config", NULL),
+			       SHELL_CMD(reset, &sub_ship_reset, "Reset button config", NULL),
+			       SHELL_CMD(mode, &sub_ship_mode, "Set ship mode", NULL),
+			       SHELL_SUBCMD_SET_END);
+
+/* Creating subcommands (level 1 command) array for command "npmx". */
+SHELL_STATIC_SUBCMD_SET_CREATE(
+	sub_npmx, SHELL_CMD(charger, &sub_charger, "Charger", NULL),
+	SHELL_CMD(buck, &sub_buck, "Buck", NULL), SHELL_CMD(ldsw, &sub_ldsw, "LDSW", NULL),
+	SHELL_CMD(leds, &sub_leds, "LEDs", NULL), SHELL_CMD(gpio, &sub_gpio, "GPIOs", NULL),
+	SHELL_CMD(adc, &sub_adc, "ADC", NULL), SHELL_CMD(pof, &sub_pof, "POF", NULL),
+	SHELL_CMD(timer, &sub_timer, "Timer", NULL),
+	SHELL_CMD(errlog, &sub_errlog, "Reset errors logs", NULL),
+	SHELL_CMD(vbusin, &sub_vbusin, "VBUSIN", NULL), SHELL_CMD(ship, &sub_ship, "SHIP", NULL),
```

Comment:
Unless it is forced by the clang, I would leave one command per line, because it is more readable.

Reply:
`pre-commit` is forcing it, I can not leave it.